src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -321,6 +343,30 @@
       result
       (recur db-spec))))
 
+(defn garbage-collector
+  [db clean-lock config]
+  (let [seconds-pos? (comp pos? to-seconds)
+        what (filter identity
+                     [(when (some-> (:node-ttl config) seconds-pos?)
+                        "expire_nodes")
+                      (when (some-> (:node-purge-ttl config) seconds-pos?)
+                        (let [limit (:node-purge-gc-batch-limit config)]
+                          (if (zero? limit)
+                            "purge_nodes"
+                            ["purge_nodes" {:batch_limit limit}])))
+                      (when (some-> (:report-ttl config) seconds-pos?)
+                        "purge_reports")
+                      "other"])]
+    (fn []
```

Comment:
Ahh, I think one reason was because the function needs that logic and to capture that state, and so this arrangement facilitates testing.